### PR TITLE
More pylint fixes

### PR DESCRIPTION
**Pull request by asankov**

### CODE REVIEW

File: tcms/core/views/index.py [hunk 1]
```diff
@@ -4,7 +4,7 @@
 from django.shortcuts import render
 
 
-def index(request, template_name='index.html'):
+def index(request, _template_name='index.html'):
```
Comment (atodorov) [inline tcms/core/views/index.py]: This parameter is completely unused so remove it instead of renaming.

File: tcms/issuetracker/types.py [hunk 2]
```diff
@@ -5,16 +5,19 @@
     implementor class for integration specifics!
 """
 
-import os
 from urllib.parse import urlencode
-import bugzilla
-from . import bugzilla_integration
+
+import os
```
Comment (atodorov) [inline tcms/issuetracker/types.py]: Is it possible for `import os` to be above the urllib import to make the text more visually appealing ?

File: tcms/xmlrpc/api/testcase.py [hunk 3]
```diff
@@ -212,8 +212,8 @@ def add_tag(case_id, tag):
         :raises: PermissionDenied if missing *testcases.add_testcasetag* permission
         :raises: TestCase.DoesNotExist if object specified by PK doesn't exist
     """
-    t, _ = Tag.objects.get_or_create(name=tag)
-    TestCase.objects.get(pk=case_id).add_tag(t)
+    tag, _ = Tag.objects.get_or_create(name=tag)
```
Comment (atodorov) [inline tcms/xmlrpc/api/testcase.py]: Doesn't pylint complain that you are overriding the `tag` parameter ?

File: tcms/xmlrpc/api/testcase.py [hunk 4]
```diff
@@ -286,18 +286,19 @@ def create(values, **kwargs):
 
         # Add tag to the case
         for tag in string_to_list(values.get('tag', [])):
-            t, c = Tag.objects.get_or_create(name=tag)
-            tc.add_tag(tag=t)
+            tag, _ = Tag.objects.get_or_create(name=tag)
```
Comment (atodorov) [inline tcms/xmlrpc/api/testcase.py]: again, doesn't pylint complain that you are overriding/shadowing the `tag` variable from the loop condition ?

File: tcms/xmlrpc/api/testcase.py [hunk 5]
```diff
@@ -286,18 +286,19 @@ def create(values, **kwargs):
...
 
-    result = tc.serialize()
-    result['text'] = tc.latest_text().serialize()
+    result = test_case.serialize()
+    result['text'] = test_case.latest_text().serialize()
 
     return result
 
 
+# pylint: disable=redefined-builtin
```
Comment (atodorov) [inline tcms/xmlrpc/api/testcase.py]: I think this is better to be placed after the function name. However you have to do little experimenting.

From:
https://docs.pylint.org/en/1.8/faq.html#is-it-possible-to-locally-disable-a-particular-message
```at the desired block level or at the end of the desired line of code```. The way I understand it is that the disable placed above the function name will turn off the redefined-builtin check until the end of the module, not only for that function.

A simple way to test this is to deliberately introduce another function which will raise warnings (even the same warning) and see which positioning kills warnings for both functions. We still want to disable checks very cautiously.

Comment (asankov) [reply tcms/xmlrpc/api/testcase.py]: Here are the results of my experiment:
I left `#pylint: disable=redefined-builtin` on top of the function signature as it is here, and defined two methods with names that shadow builtin function below that - and pylint ignored them

However, when I put `#pylint: disable=redefined-builtin` next to the function signature it caught the two other below, so it is safe to put in next to the function definition, e.g. 
`def filter(query): # pylint: disable=redefined-builtin`

File: tcms/xmlrpc/api/testcase.py (hunk 4 above)
Comment (asankov) [reply tcms/xmlrpc/api/testcase.py]: Same as above

File: tcms/xmlrpc/api/testcase.py (hunk 3 above)
Comment (asankov) [reply tcms/xmlrpc/api/testcase.py]: No, it doesn't. I suppose it thinks that I do not shadow it, but rather give another value to the parameter that is being passed.

**Pull request merged**